scintillometry/io/psrfits/hdu.py

```diff
@@ -212,6 +294,12 @@ def frequency(self):
 
         return freqs
 
+    @frequency.setter
+    def frequency(self, freqs):
+        freqs = freqs.to_value(u.MHz)
+        self.hdu.data['DAT_FREQ'] = np.broadcast_to(freq, (self.nrow,
```

Comment:
Here, the frequency will come in with shape `nchan, 1`, i.e., it will have a trailing shape of 1 (for the phase bin axis). So, maybe just act on `freq.squeeze()` (or explicitly check the shape).